File: JavaScript/2500_delete_greater_value_in_each_row/solution.js
function delete_greatest_in_row(grid) {
    let result = [];
    while (grid[0].length != 0) {
        let get_max = [];
        for (let row of grid) {
            const max = Math.max(...row);
            get_max.push(max);
            row.splice(row.indexOf(max), 1);
        }
        result.push(Math.max(...get_max));
    }
    return result.reduce((a,b) => a + b, 0)
}

let grid = [[1,2,4],[3,3,1]];
console.log(delete_greatest_in_row(grid));

File: Python/2500_delete_greater_value_in_each_row/solution.py
def delete_greatest_in_row(grid):
    result = list()
    while len(grid[0]) != 0:
        get_max = []
        for i in grid:
            get_max.append(max(i))
            i.remove(max(i))
        result.append(max(get_max))
        
    return sum(result)

grid = [[1,2,4],[3,3,1]]
print(delete_greatest_in_row(grid))